pkgs/http_client_conformance_tests/lib/src/redirect_tests.dart

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
```

Comment:
Should we try to make these work for testing web implementation of clients as well?

If we want this to be cross platform we'd need to use a `spawnHybrid` call to run the server in the test runner VM.
It would require defining how we'd control the server behavior over a stream channel, I haven't though much yet about a design for such an API.

See https://github.com/dart-lang/http/pull/700
In that case it was easy because the server behavior is static and doesn't interact directly with the test other than through the HTTP calls so we didn't need to do anything but communicate the port.